app/Model/AppModel.php

```diff
@@ -849,7 +849,184 @@ public function delete($id = null, $cascade = true) {
     
     return $ret;
   }
-  
+
+  /**
+   * Duplicate objects.
+   *
+   * @since  COmanage Registry v3.2.0
+   * @param  Model   $model      Model to duplicate
+   * @param  String  $foreignKey The name of the foreign key from $model to its parent (or 'id' if no parent)
+   * @param  Integer $fkid       The ID of the foreign key in $model
+   * @param  Array   $idmap      Reference to an array of old IDs to new IDs (originals to duplicates)
+   * @param  Boolean $isTree     True if $model implements TreeBehavior
```

Comment:
I realize this is just moving code old around, but we probably should have done something like `$model->Behaviors->loaded('Tree')` to test for TreeBehavior instead of having to explicitly pass a parameter...

Reply:
Done